### PR TITLE
PLATY-551: Rejecting files if MIME type not configured for service

### PR DESCRIPTION
**Pull request by JoPintoPaul**

### CODE REVIEW

**Comment by konradwudkowski** [comment]
@JoPintoPaul it looks like `quarantined` has been replaced by `rejected` sometimes but there are still many references to `quarantine` in test names, config (quarantineBucket) etc. Is there a need to keep both?

**Comment by JoPintoPaul** [comment]
@konradwudkowski That's a good point, I should remove the references to `quarantined` throughout. I'll work on that and let you know once it's completed.

File: app/services/FileCheckingResultHandler.scala [hunk 1]
```diff
@@ -35,24 +35,34 @@ class FileCheckingResultHandler @Inject()(fileManager: FileManager, virusNotifie
...
       _ <- fileManager.delete(objectLocation)
     } yield ShouldTerminate
 
-  private def handleValid(objectLocation: S3ObjectLocation)(implicit ec: ExecutionContext) =
+  private def handleIncorrectType(objectLocation: S3ObjectLocation, mimeType: MimeType, serviceName: Option[String])(
+    implicit ec: ExecutionContext) =
     for {
-      _ <- fileManager.copyToOutboundBucket(objectLocation)
+      metadata <- fileManager.getObjectMetadata(objectLocation)
+      details                 = s"MIME type [${mimeType.value}] is not allowed for service: [${serviceName.getOrElse("No service name provided")}]"
```
Comment (konradwudkowski) [inline app/services/FileCheckingResultHandler.scala]: it's not wrong but `details` and `quarantineObjectContent` could be outside/above the for comprehension, formatting would also look better. As mentioned in a comment why not call the object `rejectedObjectContent`?

File: app/services/FileTypeCheckingService.scala [hunk 2]
```diff
@@ -31,19 +31,30 @@ class FileTypeCheckingService @Inject()(fileTypeDetector: FileTypeDetector, serv
     location: S3ObjectLocation,
     objectContent: ObjectContent,
     objectMetadata: ObjectMetadata): Future[FileCheckingResult] = {
-    val fileType = fileTypeDetector.detectType(objectContent.inputStream, objectMetadata.originalFilename)
+    val fileType         = fileTypeDetector.detectType(objectContent.inputStream, objectMetadata.originalFilename)
```
Comment (konradwudkowski) [inline app/services/FileTypeCheckingService.scala:35]: method is not async, does the return type need to be a future?

File: test/services/FileTypeCheckingServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+      val configuration = mock[ServiceConfiguration]
+      when(configuration.consumingServicesConfiguration).thenReturn(consumingServicesConfiguration)
+
+      val checkingService = new FileTypeCheckingService(detector, configuration)
+
+      When("the file is checked")
+      val result: FileCheckingResult = Await.result(checkingService.check(location, content, metadata), 2.seconds)
+
+      Then("a valid result should be returned")
```
Comment (konradwudkowski) [inline test/services/FileTypeCheckingServiceSpec.scala]: is it a *valid* result here? Should the test say invalid/incorrect etc?

File: test/services/FileTypeCheckingServiceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
```
Comment (konradwudkowski) [inline test/services/FileTypeCheckingServiceSpec.scala]: is it a *valid* result here? Should the test say invalid/incorrect etc?

File: test/services/FileTypeCheckingServiceSpec.scala (hunk 3 above)
Comment (JoPintoPaul) [reply test/services/FileTypeCheckingServiceSpec.scala]: I have corrected this and pushed to the branch.

File: app/services/FileTypeCheckingService.scala (hunk 2 above)
Comment (JoPintoPaul) [reply app/services/FileTypeCheckingService.scala:35]: This is to do with how we chain file scans, to provide a consistent way of handling multiple checks that a file needs to pass. I've added in a Trait that makes this relationship explicit.

File: app/services/FileCheckingResultHandler.scala (hunk 1 above)
Comment (JoPintoPaul) [reply app/services/FileCheckingResultHandler.scala]: Corrected and pushed to the branch.

**Comment by JoPintoPaul** [comment]
@konradwudkowski I have reverted all references to "rejected" and left them as "quarantined" - we've decided that we'll do all this renaming through our multiple projects and infra at the same time as a separate ticket.

**Review by konradwudkowski — APPROVED** [review]
:+1:

**Pull request merged**